tests/test-utils.js

```diff
@@ -0,0 +1,81 @@
+import path from 'path';
+import webpack from 'webpack';
+import Promise from 'bluebird';
+import MemoryFs from 'memory-fs';
+
+const fs = new MemoryFs();
+const unitTestFixtures = path.resolve(__dirname, 'fixtures');
+
+/**
+ *
+ *
+ * @param {string} fixtureName
+ * @param {string} [withQueryString='']
+ * @returns {string} Absolute path of a file with query that is to be run by a loader.
+ */
+function getFixtureResource(fixtureName, withQueryString = '') {
+  return `${getFixture(fixtureName)}?${withQueryString}`;
+}
+
+/**
+ *
+ *
+ * @param {string} fixtureName
+ * @returns {string} Absolute path of a file with query that is to be run by a loader.
+ */
+function getFixture(fixtureName) {
+  return path.resolve(unitTestFixtures, fixtureName);
+}
+
+/**
+ *
+ *
+ * @param {Object} withOptions - Loader Options
+ * @returns {{loader: string, options: Object}}
+ */
+function getLoader(withOptions) {
+  return [{ loader: path.resolve(__dirname, '../src/index.js'), options: withOptions }];
+}
+
+/**
+ *
+ *
+ * @param {string} exampleName
+ * @returns {Object|Array} - Returns an object or array of objects representing the webpack configuration options
+ */
+function getExampleConfig(exampleName) {
+  return require(`../examples/${exampleName}/webpack.config.js`); //eslint-disable-line
+}
+
+/**
+ *
+ *
+ * @param {string} exampleName - name of example inside of examples folder
+ * @returns
+ */
+async function runWebpackExampleInMemory(exampleName) {
+  const webpackConfig = getExampleConfig(exampleName);
+  const compiler = webpack(webpackConfig);
+
+  compiler.outputFileSystem = fs;
+
+  const run = Promise.promisify(compiler.run, { context: compiler });
+  const stats = await run();
+
+  const { compilation } = stats;
+  const { errors, warnings, assets, entrypoints, chunks, modules } = compilation;
```

Comment:
Destructuring here and returning an {Object} on L69 doesn't really make sense, its better to return the whole `compilation` and pass it to helpers  in the particular test

**example.test.js**
```js
import webpack from 'helpers/compiler'
import { loader, errors, warnings, assets } from 'helpers/utils'  // maybe test-utils/stats-utils

test('Loader', () => {
   const config = { loader: {...options} }

   webpack('path/to/fixture.ext', config)
    .then((compilation) => loader(compilation))
    .then({ result, map, meta } => {
       expect(result).toMatchSnapshot()
       if (map) expect(map).toMatchSnapshot()
       ...
    })
   .catch((err) => err)
})
```

Reply:
But thats discussion for [webpack-defaults #68](https://github.com/webpack-contrib/webpack-defaults/issues/68) to not derail the PR here further :)